src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -666,21 +666,32 @@ Status CoreWorkerDirectTaskSubmitter::CancelTask(TaskSpecification task_spec,
       [this, task_spec, scheduling_key, force_kill, recursive](
           const Status &status, const rpc::CancelTaskReply &reply) {
         absl::MutexLock lock(&mu_);
+        RAY_LOG(DEBUG) << "CancelTask RPC response received for " << task_spec.TaskId()
+                       << " with status " << status.ToString();
         cancelled_tasks_.erase(task_spec.TaskId());
-
-        if (status.ok() && !reply.attempt_succeeded()) {
-          if (cancel_retry_timer_.has_value()) {
-            if (cancel_retry_timer_->expiry().time_since_epoch() <=
-                std::chrono::high_resolution_clock::now().time_since_epoch()) {
-              cancel_retry_timer_->expires_after(boost::asio::chrono::milliseconds(
-                  RayConfig::instance().cancellation_retry_ms()));
+        if (status.ok()) {
```

Comment:
Is it clearer to keep the `status.ok() && !reply.attempt_succeeded()` condition? We can remove one level of condition nesting in each branch.